### ISSUE TITLE
Remove `target_spacing_sufficient` from playwright config exclusions

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#6365]
In one of the recent accessibility-checker rulesets, `target_spacing_sufficient` was now causing failures. See original [Slack thread](https://ibm-studios.slack.com/archives/C036P1CTN/p1727885751674519).

Review if we can remove `target_spacing_sufficient` from our playwright exclusions or if we need to open followup design issues to address target size issues.

### Acceptance criteria
- [x] Remove `target_spacing_sufficient` from `playwright.config.js`
- [ ] Document or playback the current violations to the team
